diff --git a/setup.py b/setup.py
index 676beb6..13b50c8 100644
--- a/setup.py
+++ b/setup.py
@@ -11,7 +11,7 @@ extensions = [
         "pguresvt._pguresvt",
         sources=["pguresvt/_pguresvt.pyx"],
         include_dirs=["pguresvt/", "src/", np.get_include()],
-        libraries=["openblas", "lapack", "armadillo", "nlopt"],
+        libraries=["lapack", "armadillo", "nlopt"],
         language="c++",
         extra_compile_args=[
             "-fPIC",

